test/RelayHubGasCalculations.test.ts

```diff
@@ -337,6 +345,88 @@ contract('RelayHub gas calculations', function ([_, relayOwner, relayWorker, rel
       })
   })
 
+  describe('relayCall()\'s msg.data cost calculations', function () {
+    enum RelayCallDynamicArg {
+      APPROVAL_DATA = 'approvalData',
+      ENCODED_FUNCTION = 'encodedFunction',
+      PAYMASTER_DATA = 'paymasterData'
+    }
+
+    before(async function () {
+
+    });
+    [RelayCallDynamicArg.APPROVAL_DATA, RelayCallDynamicArg.ENCODED_FUNCTION, RelayCallDynamicArg.PAYMASTER_DATA].forEach(dynamicArg => {
+      const gassesUsed: any[] = [];
+      [0, 32, 128, 16384/* , 32768, 65536 */].forEach(dataLength => {
+        it(`with arg: ${dynamicArg} length: ${dataLength}`, async function () {
+          // console.log('gasUsed: ', gassesUsed)
+          const id = (await snapshot()).result
+          const senderNonce = (await forwarderInstance.getNonce(senderAddress)).toString()
+          let approvalData = '0x'
+          let paymasterData = '0x'
+          let encodedFunction = recipient.contract.methods.dontEmitMessage('').encodeABI()
+          if (dynamicArg === RelayCallDynamicArg.APPROVAL_DATA) {
+            approvalData = '0x' + 'ff'.repeat(dataLength)
+          } else if (dynamicArg === RelayCallDynamicArg.ENCODED_FUNCTION) {
+            // encoded function arg is an ascii string, so each char is a byte. We append half the length then.
```

Comment:
I don't see what 'half the length' refers to here

Reply:
Removed the comment anyway